perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -41,6 +43,405 @@
 ALLOW_ALL = 'tcp:1-65535,udp:1-65535,icmp'
 
 
+class GceVPNGW(network.BaseVPNGW):
+  CLOUD = providers.GCP
+
+  def __init__(self, name, network_name, region, cidr, project):
+    super(GceVPNGW, self).__init__()
+
+    self.forwarding_rules = {}
+    self.tunnels = {}
+    self.routes = {}
+    self.ip_resource = None
+    self.vpn_gw_resource = GceVPNGWResource(name, network_name, region, cidr, project)
+
+    self.name = name
+    self.network_name = network_name
+    self.region = region
+    self.cidr = cidr
+    self.project = project
+
+    self.ip_address = None
+    self.created = False
+    self.require_target_to_init = False
+    self.routing = None
+    self.psk = None
+
+  def ConfigureTunnel(self, tunnel_config):
+    network.BaseVPNGW.ConfigureTunnel(self, tunnel_config)
+    logging.info('Configuring Tunnel with params:')
+    logging.info(tunnel_config)
+
+    # update tunnel_config if needed
+    if self.name not in tunnel_config.endpoints:
+      logging.info('tunnel_config: This endpoint isnt registered yet... %s' % self.name)
+      tunnel_config.endpoints[self.name] = {}
+      tunnel_config.endpoints[self.name]['is_configured'] = False
```

Comment:
Easier to read if you create the dict from the start:
  tunnel_config.endpoints[self.name] = {
   'is_configured' : False,
  'cidr': self.cidr,
  ...
}

Reply:
Fixed.